pkg/microservice/systemconfig/core/codehost/service/codehost.go

```diff
@@ -69,3 +78,114 @@ func UpdateCodeHostByToken(host *models.CodeHost, _ *zap.SugaredLogger) (*models
 func GetCodeHost(id int, _ *zap.SugaredLogger) (*models.CodeHost, error) {
 	return mongodb.NewCodehostColl().GetCodeHostByID(id)
 }
+
+type state struct {
+	CodeHostID  int    `json:"code_host_id"`
+	RedirectURL string `json:"redirect_url"`
+}
+
+func AuthCodeHost(redirectURI string, codeHostID int, logger *zap.SugaredLogger) (string, error) {
+	codeHost, err := GetCodeHost(codeHostID, logger)
+	if err != nil {
+		logger.Errorf("GetCodeHost:%s err:%s", codeHostID, err)
+		return "", err
+	}
+	redirectParsedURL, err := url.Parse(redirectURI)
+	if err != nil {
+		logger.Errorf("Parse redirectURI:%s err:%s", redirectURI, err)
+		return "", err
+	}
+	callbackURL := url.URL{
```

Comment:
Why not reuse `redirectParsedURL`?

Reply:
func Parse(rawURL string) (*URL, error) {  it's a pointer , reuse may modify the redirectParsedURL?

Reply:
`redirectParsedURL` only plays the role of state data in the future, so will reuse it affect the subsequent semantics?

Reply:
changed